commands/tocfl/chewing.py

```diff
@@ -0,0 +1,262 @@
+# Neutral tones are not labeled
+DIACRITIC_TO_BASE_AND_TONE = {
+    # ā (ɑ̄) ē ī ō ū ǖ
+    # á (ɑ́) é í ó ú ǘ
+    # ǎ (ɑ̌) ě ǐ ǒ ǔ ǚ
+    # à (ɑ̀) è ì ò ù ǜ
+    "ā": ("a", "¯"),
+    "á": ("a", "ˊ"),
+    "ǎ": ("a", "ˇ"),
+    "ă": ("a", "ˇ"),
+    "à": ("a", "ˋ"),
+    "ē": ("e", "¯"),
+    "é": ("e", "ˊ"),
+    "ě": ("e", "ˇ"),
+    "ĕ": ("e", "ˇ"),
+    "è": ("e", "ˋ"),
+    "ī": ("i", "¯"),
+    "í": ("i", "ˊ"),
+    "ǐ": ("i", "ˇ"),
+    "ĭ": ("i", "ˇ"),
+    "ì": ("i", "ˋ"),
+    "ō": ("o", "¯"),
+    "ó": ("o", "ˊ"),
+    "ǒ": ("o", "ˇ"),
+    "ŏ": ("o", "ˇ"),
+    "ò": ("o", "ˋ"),
+    "ū": ("u", "¯"),
+    "ú": ("u", "ˊ"),
+    "ǔ": ("u", "ˇ"),
+    "ŭ": ("u", "ˇ"),
+    "ù": ("u", "ˋ"),
+    "ǖ": ("ü", "¯"),
+    "ǘ": ("ü", "ˊ"),
+    "ǚ": ("ü", "ˇ"),
+    "ü̆": ("ü", "ˇ"),
+    "ǜ": ("ü", "ˋ"),
+    "ề": ("ê", "ˋ"),  # 「ㄝ」 could only possibly be the fourth tone
+}
+
+
+def to_chewing(pinyin: str) -> str:
+
+    # Matches chewing from substrings
+    def match_chewing(string: str, index: int, target: dict[str, str]):
+        global PINYIN_COMBINED, PINYIN_FINALS
+        # Substrings only to the maxium possible character amount
+        for i in range(max([len(i) for i in target.keys()]), 0, -1):
+            target_str = string[index : index + i]
+            result = target.get(target_str)
+            if result:
+                # Resolve duplicates
+                if target == PINYIN_COMBINED:
+                    if target_str == "uan" and string[index - 1] in [
+                        "y",
+                        "j",
+                        "q",
+                        "x",
+                    ]:
+                        result = "ㄩㄢ"
+                    elif target_str == "un" and string[index - 1] in [
+                        "y",
+                        "j",
+                        "q",
+                        "x",
+                    ]:
+                        result = "ㄩㄣ"
+                elif target == PINYIN_FINALS:
+                    if target_str == "e" and string[index - 1] == "y":
+                        result = "ㄝ"
+
+                return (index + i, result)
+        return (index + 1, None)
+
+    # Ensure there are no trailing characters unable to form word
+    def forms_new_word(index: int):
+        nonlocal pinyin
+        global PINYIN_INITIALS, PINYIN_ALONE
+        return (
+            match_chewing(pinyin, index, PINYIN_INITIALS | PINYIN_ALONE)[1]
+            or index >= len(pinyin)
+            or not pinyin[index].isalpha()
+        )
+
+    # Temporarily store the chewing tones and original index
+    tones = []
+    for index in range(len(pinyin)):
+        value = DIACRITIC_TO_BASE_AND_TONE.get(pinyin[index])
+        if not value:
+            continue
+        pinyin = pinyin[:index] + value[0] + pinyin[index + 1 :]
+        tones.append((index, value[1]))
+
+    chewing = ""
+    index = 0
+    while index < len(pinyin):
```

Comment:
This part is fine but I think adding an example as comments would help explain what's going on, also very useful when you write tests and a bug is found and you need to debug this part later

So like 

```py
# example word: tái
if (condition):
  chewing += something # ㄊ
# some other code
chewing += something # ㄊㄞ
# add tone
chewing += something # ㄊㄞˊ
```